core/src/c/offheap_jni_JemallocWrapper.cpp

```diff
@@ -0,0 +1,25 @@
+#include "offheap_jni_JemallocWrapper.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+JNIEXPORT jlong JNICALL Java_offheap_jni_JemallocWrapper_malloc_10 (JNIEnv *, jclass, jlong size) {
+    const auto result = je_malloc(static_cast<size_t>(size));
+
+    return result == NULL ? -1 : (jlong) result;
+}
+
+JNIEXPORT jlong JNICALL Java_offheap_jni_JemallocWrapper_realloc_10 (JNIEnv *, jclass, jlong address, jlong newSize) {
+    const auto result = je_realloc((void *) address, static_cast<size_t>(newSize));
```

Comment:
Is `size_t` guaranteed to be 64-bit on x86-64? What about x86? 

`s.m.Unsafe` always uses longs for addresses and sizes but it's assumed to be in 32-bit range on 32-bit platforms. I think that we'll need some extra machinery to cast this safely.